mdaviz/mda_file_table_view.py

```diff
@@ -0,0 +1,179 @@
+
+    def data(self):
+        """Return the data from the table view:
+              self.data= fileInfo, fields
+        with:
+        - fileInfo (dict): A dictionary of all the file information:
+            - fileName (str): The name of the file without its extension.
+            - filePath (str): The full path of the file.
+            - folderPath (str): The full path of the parent folder.
+            - metadata (dict): The extracted metadata from the file.
+            - scanDict (dict): A dictionary of positioner & detector dataset for plot.
+            - firstPos (float): The first positioner (P1, or if no positioner, P0 = index).
+            - firstDet (float): The first detector (D01).
+            - pvList (list of str): List of detectors PV names as strings.
+        - field (list): List of TableField object, one for each det/pos:
+                ([TableField(name='P0', selection=None,...
+                ...desc='Index', pv='Index', unit='a.u'),...])
+        """
+        return self._data
+
+    def setData(self):
+        self._data = {}
+        if self.mda_file.data() != {}:
+            fileInfo = self.mda_file.data()
+            scanDict = self.mda_file.data()["scanDict"]
+            fields = [
+                TableField(
+                    v["fieldName"],
+                    selection=None,
+                    pv=v["name"],
+                    desc=v["desc"],
+                    unit=v["unit"],
+                )
+                for v in scanDict.values()
+            ]
+        self._data = {"fileInfo": fileInfo, "fields": fields}
+
+    def displayTable(self, selection_field):
+        from .mda_file_table_model import MDAFileTableModel
+        from .empty_table_model import EmptyTableModel
+
+        if self.data() is not None:
+            fields = self.data()["fields"]
+            data_model = MDAFileTableModel(
+                COLUMNS, fields, selection_field, self.mda_file.mda_mvc
+            )
+            self.tableView.setModel(data_model)
+            # Hide Field/Mon/Norm columns (Field = vertical header, Mon & Norm not yet implemented)
+            for i in [0, 3, 4]:
+                self.tableView.hideColumn(i)
+        else:
+            # No MDA files to display, show an empty table with headers
+            empty_model = EmptyTableModel(HEADERS)
+            self.tableView.setModel(empty_model)
+
+    def setStatus(self, text):
+        self.mda_file.mda_mvc.setStatus(text)
+
+    def clearContents(self):
+        self.tableView.model().clearAllCheckboxes()
+        self.tableView.setModel(None)
+
+    def data2Plot(self, selections):
+        """
+        Extracts selected datasets for plotting from scanDict based on user selections.
+
+        Parameters:
+        - selections: A dictionary with keys "X" and "Y", where "X" is the index for the x-axis data
+        and "Y" is a list of indices for the y-axis data.
```

Comment:
Restructured text indentation problems here and elsewhere similar.  The additional lines below a `-` should be indented the same as the first text on the line with the `-`.  Might be flagged as an error at the time sphinx builds the HTML pages.